backend/api/badge_scanner/schema.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pretix
+import re
+from typing import Any
+import strawberry
+from strawberry.types.info import Info
+from typing import Optional, Union
+
+from api.permissions import IsAuthenticated
+from badge_scanner import models
+from conferences.models import Conference
+from users.client import get_user_by_email
+
+
+@strawberry.type
+class ScanError:
+    message: str
+
+
+@strawberry.input
+class ScanBadgeInput:
+    url: str
+    conference_code: str
+
+    url_regex = re.compile(r"^https://pycon\.it/b/([\w\d-]+)$")
+
+    def validate(self) -> ScanError | None:
+        if not self.url:
+            return ScanError(message="URL is required")
+
+        match = self.url_regex.match(self.url)
+
+        return None if match else ScanError(message="URL is not valid")
+
+    @property
+    def order_position_id(self) -> str:
+        match = self.url_regex.match(self.url)
+
+        assert match
+
+        return match.group(1)
+
+
+@strawberry.type
+class Attendee:
+    full_name: str
```

Comment:
```suggestion
    fullname: str
```

I need to double-check, but I think in all places we are now preferring `fullname` instead of `fullName`

Reply:
<img width="1308" alt="CleanShot 2023-05-10 at 10 46 56@2x" src="https://github.com/pythonitalia/pycon/assets/667029/9131ed4b-45c6-4e53-b138-842ab55c329f">

we use fullName more, happy to change it to fullname though

Reply:
Is this how many times the field is called or the number of time a field named `fullName` exists in the API? I think in all my APIs I have used `fullname` now

Reply:
how many times the field is called